src/librustc_typeck/collect.rs

```diff
@@ -48,9 +48,48 @@ use rustc::hir::GenericParamKind;
 use rustc::hir::{self, CodegenFnAttrFlags, CodegenFnAttrs, Unsafety};
 
 use std::iter;
+use std::ops::Range;
 
 struct OnlySelfBounds(bool);
 
+pub fn lint_duplicate_auto_traits<'a, 'gcx, 'tcx>(
```

Comment:
I see no particular reason to limit this to auto traits, but I am also fine with just linting auto traits for now.